Google.PowerShell/Container/GkeClusterCmdlets.cs

```diff
@@ -487,6 +491,72 @@ protected NodeConfig BuildNodeConfig()
 
             return nodeConfig;
         }
+
+        /// <summary>
+        /// Helper function to build a NodePool object.
+        /// InitialNodeCount will default to 1.
+        /// If AutoScaling is true, MaximumNodesToScaleTo has to be specified.
+        /// MaximumNodesToScaleTo has to be greater than MinimumNodesToScaleTo, which defaults to 1.
+        /// </summary>
+        /// <param name="name"></param>
+        /// <param name="config"></param>
+        /// <param name="initialNodeCount"></param>
+        /// <param name="autoUpgrade"></param>
+        /// <param name="autoScaling"></param>
+        /// <param name="minimumNodesToScaleTo"></param>
+        /// <param name="maximumNodesToScaleTo"></param>
+        /// <returns></returns>
+        protected NodePool BuildNodePool(string name, NodeConfig config, int? initialNodeCount, bool autoUpgrade,
+            bool autoScaling, int? minimumNodesToScaleTo, int? maximumNodesToScaleTo)
+        {
+            var nodePool = new NodePool()
+            {
+                Name = name,
+                InitialNodeCount = initialNodeCount ?? 1,
+                Config = config
+            };
+
+            if (autoScaling)
+            {
+                var scaling = new NodePoolAutoscaling() { Enabled = true };
+                if (maximumNodesToScaleTo == null)
+                {
+                    throw new PSArgumentException(
```

Comment:
I would consider using an overload that also takes min and max nodesToScale, rather than including the boolean parameter autoScaling.